### ISSUE TITLE
Crestron closes connection 

### ISSUE THREAD

**Issue by DasSauerkraut89** [songzh96/node-red-contrib-crestron#18]
If I send about 10 values every 5 seconds to  Crestron, Crestron closes the connection to the node. A reconnect isnt possible, i have to restart the Crestron Program via Toolbox. If I send the 10 values, let me say, every 60 Minutes the Connection is stable. Do you have an idea how to fix this issue?

"Error: connect ECONNREFUSED 192.168.2.13:9999"

**Comment by songzh96** [comment]
Are you using a 4 series console?
Sometimes, updates to the simpl software can cause problems. I don't have a test environment right now, so it's hard to fix this bug, I suggest you downgrade the software version




Songzh
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "songzh96/node-red-contrib-crestron"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年5月18日(星期四) 下午3:17
***@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;[songzh96/node-red-contrib-crestron] Crestron closes connection  (Issue #18)





 
If I send about 10 values every 5 sekunds Cresteon closes the connection to the node. A reconnect isnt possible, i have to restart the Crestron Program via Toolbox. If I send the 10 values, let me say, every 60 Minutes the Connection is stable. Do you have an idea how to fix this issue?
 
"Error: connect ECONNREFUSED 192.168.2.13:9999"

**Comment by DasSauerkraut89** [comment]
I have tested some szenarios during the last weeks. I never had a stable connection. Today i looked in the Crestron Error Log of the DIN-AP4. I found this entry:

149. Notice: ctpd # 2023-07-30 19:25:14 # SHELL Connection 1 from: 192.168.1.150 port 63132
  150. Error: SimplPlus[App01] # 2023-07-30 19:25:33 # Module S-11.1 : CresNode at line 211: Unhandled Exception: System.OutOfMemoryException: Insufficient memory to continue the execution of the program.
  at (wrapper alloc) System.Object.AllocString(intptr,int)
  at System.Text.StringBuilder.ToString () [0x
  151. Error: SimplPlus[App01] # 2023-07-30 19:25:33 # 00014] in <2ad40006e9c141b299c9c7c533021a74>:0 
  at Crestron.Logos.SplusObjects.CrestronString..ctor (System.Text.StringBuilder stringValue, Crestron.Logos.SplusObjects.CrestronStringEncoding uiEncoding, System.Int32 maxCapacity, Crestron.Logos.SplusObje

The DIN-AP4 hast 4GB of RAM. I Always get the error according to your module. Can you check if there is anything in the code which completely fill up the RAM? Reading the TCP Buffer is sometimes strange with Crestrong devices.

I'm using Firmware 2.8001.00086 on the DIN-AP4

**Comment by galaye** [comment]
still does not reconect ...:
"Error: connect ECONNREFUSED 192.168.1.37:9999"

bummer

**Comment by lonestar6262** [comment]
Dear Songzh,

Could you open up the Crestron module by publishing the password - I even would spend some money for your work.

Best regards

Ralf

**Comment by gregjwinter** [comment]
Yes Zongzh – I would do the same.

**Comment by cismarine** [comment]
> Dear Songzh,
> 
> Could you open up the Crestron module by publishing the password - I even would spend some money for your work.
> 
> Best regards
> 
> Ralf

Send me a message, I might be able to help. unmoved.fob.0a@icloud.com

**Comment by DasSauerkraut89** [comment]
Hello,I can give you the password of the Module, if the author isnt answering!Greetings from germany,Marcel Von meinem iPhone gesendetAm 11.01.2024 um 17:52 schrieb cismarine ***@***.***>:﻿

Dear Songzh,
Could you open up the Crestron module by publishing the password - I even would spend some money for your work.
Best regards
Ralf

Send me a message, I might be able to help. ***@***.***

**Comment by lonestar6262** [comment]
Hi Marcel,
what e-mail to use?
BR 
Ralf

**Comment by songzh96** [comment]
Hi，Ralf


Sorry for the late reply due to time difference


How many nodes do you need?
You need to give me the source files and I will help you implement your needs



Best Regards
Songzh






Songzh
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "songzh96/node-red-contrib-crestron"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年1月12日(星期五) 凌晨0:52
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [songzh96/node-red-contrib-crestron] Crestron closes connection  (Issue #18)





  
Dear Songzh,
 
Could you open up the Crestron module by publishing the password - I even would spend some money for your work.
 
Best regards
 
Ralf
  
Send me a message, I might be able to help. ***@***.***

**Comment by lonestar6262** [comment]
Hi Songzh,

basically I have to replicate the Crestron App Module which has the follwing nodes towards the Node Red
  
  Digital  = 128
  Analog =   68
  Serial = 32

If you can build such a crestron module with the same look / functionality this would be great

Thanks in advance

Ralf

**Comment by songzh96** [comment]
Hi Ralf


sorry for the late reply


I have two payment plans here, you can choose any one


1. 110$ I will customize it according to your requirements


2. For 330$ you can get the source code and I will remove the password and send it to you.


Best Regards
SongZh





Songzh
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "songzh96/node-red-contrib-crestron"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年1月12日(星期五) 下午5:20
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [songzh96/node-red-contrib-crestron] Crestron closes connection  (Issue #18)





 
Hi Songzh,
 
basically I have to replicate the Crestron App Module which has the follwing nodes
 
Digital  = 128
 Analog =   68
 Serial = 32
 
If you can build such a crestron module with the same look / functionality this would be great
 
Thanks in advance
 
Ralf

**Comment by lonestar6262** [comment]
Hi Songzh,

Thanks for your offer - could you customize the Crestron node module that it has three tabs  
  -digital
  -analog
  -serial

and one more question, what does the digital false - can I use it as a digital input as digital true?

What do I have to put in in the node on the node red side that‘s not clear for me.

Best regards

Ralf

**Comment by gregjwinter** [comment]
Thank you for the offer – I use this code for my house and want to maintain it for the future.  Does PayPal still work for payment of $330?

**Comment by songzh96** [comment]
Hi Ralf


May I ask, are you a service provider or a consumer?
If you are a service provider, maybe we can cooperate


If you are a consumer and I can help you write the program you want, or if you want to program it yourself, you need to tell me what you want to know.


Because I rarely read emails, I can’t reply in time



Best Regards
Songzh


Songzh
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "songzh96/node-red-contrib-crestron"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年1月19日(星期五) 凌晨4:16
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [songzh96/node-red-contrib-crestron] Crestron closes connection  (Issue #18)





 
Hi Songzh,
 
Thanks for your offer - could you customize the Crestron node module that it has three tabs
 -digital
 -analog
 -serial
 
and one more question, what does the digital false - can I use it as a digital input as digital true?
 
What do I have to put in in the node on the node red side that‘s not clear for me.
 
Best regards
 
Ralf

**Comment by songzh96** [comment]
Hi Gregory Winter


I rarely write Crestron programs now, because my company now mainly works on KNX, Aqara, and YeeLight projects.


If you support me for a fee, I will help you achieve what you want.


If you can maintain it yourself, then you only need to spend 330$, and I will send you the unencrypted module.



Best Regards
Songzh






Songzh
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "songzh96/node-red-contrib-crestron"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年1月19日(星期五) 凌晨4:28
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [songzh96/node-red-contrib-crestron] Crestron closes connection  (Issue #18)





 Thank you for the offer – I use this code for my house and want to maintain it for the future.  Does PayPal still work for payment of $330?

**Comment by lonestar6262** [comment]
Hi Songzh,

I am a consumer and not able to program in Simpl+ I am using Simpl Windows - this is why it's crucial for me to use the modules in Simpl Windows.
Your module itself works well for with the Simpl Windows Version 
SIMPL Windows:  4.17 (build: 4.1700.03)
and the DIN-AP3 Processor
As said, the only thing I am mssing is the additional IOs to connect between Crestron and Node-Red

Best regards

Ralf

**Comment by songzh96** [comment]
Hi&nbsp; Ralf


I will customize the SIMPL module according to your requirements


If you have any questions in use, please inform in time


About the issues you mentioned earlier.


DIGITAL TRUE and DIGITAL FALSE is a pulse signal, two output pins


DIGITAL Input is an input signal. When you enter it as True, the pin is pulled high, and the FALSE pin is pulled down



Best Regards
Songzh




Songzh
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "songzh96/node-red-contrib-crestron"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年1月19日(星期五) 晚上11:28
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [songzh96/node-red-contrib-crestron] Crestron closes connection  (Issue #18)





 
Hi Songzh,
 
I am a consumer and not able to program in Simpl+ I am using Simpl Windows - this is why it's crucial for me to us the modules in Simpl Windows.
 The module itself works well for with the Simpl Windows Version
 SIMPL Windows:  4.17 (build: 4.1700.03)
 and the DIN-AP3 Processor
 
Best regards
 
Ralf

**Comment by lonestar6262** [comment]
Dear Songzh,
I am looking forward to the customized module of the CresNode.
What about the payment - does PayPal work?
Thanks advance
Ralf

**Comment by songzh96** [comment]
Hi Ralf


You need to tell me what kind of module you need to customize before I can estimate. If you just change the Cresnode module, it should not cost a lot.



------------------ Original ------------------
From: lonestar6262 ***@***.***&gt;
Date: Sat,Jan 20,2024 7:22 PM
To: songzh96/node-red-contrib-crestron ***@***.***&gt;
Cc: songzh96 ***@***.***&gt;, Comment ***@***.***&gt;
Subject: Re: [songzh96/node-red-contrib-crestron] Crestron closes connection (Issue #18)





 
Dear Songzh,
 I am looking forward to the customized module of the CresNode.
 What about the payment - does PayPal work?
 Thanks advance
 Ralf

**Comment by lonestar6262** [comment]
Hi Songzh,
Yes, I just need the Crestron Module (CresNode) Updated as specified before.

Digital = 128
Analog = 68
Serial = 32

Each of the section on a separate tab.
If you have any questions, please let me know
Best regards
Ralf

**Comment by DasSauerkraut89** [comment]
Can you explain your issue? You can use the Module for every join you Need…?Von meinem iPhone gesendetAm 20.01.2024 um 17:12 schrieb lonestar6262 ***@***.***>:﻿
Hi Songzh,
Yes, I just need the Crestron Module (CresNode) Updated as specified before.
Digital = 128
Analog = 68
Serial = 32
Each of the section on a separate tab.
If you have any questions, please let me know
Best regards
Ralf

**Comment by songzh96** [comment]
Hi Ralf
Give me a private email and I will send you the customized module


Best Regards
Songzh


Songzh
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "songzh96/node-red-contrib-crestron"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年1月21日(星期天) 凌晨0:12
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [songzh96/node-red-contrib-crestron] Crestron closes connection  (Issue #18)





 
Hi Songzh,
 Yes, I just need the Crestron Module (CresNode) Updated as specified before.
 
Digital = 128
 Analog = 68
 Serial = 32
 
Each of the section on a separate tab.
 If you have any questions, please let me know
 Best regards
 Ralf

**Comment by lonestar6262** [comment]
Hi Songzh,
you can send it to ralf.aicher@berlin.de
Thanks a lot.
Where to send the money?
Thanks Ralf
